misc.go

```diff
@@ -138,8 +143,16 @@ func postForm(ctx context.Context, endpoint string, values url.Values, intf inte
 
 	// Slack seems to send an HTML body along with 5xx error codes. Don't parse it.
 	if resp.StatusCode != 200 {
+		retryAfter := 0
+		if resp.StatusCode == 429 {
+			retryAfter, _ = strconv.Atoi(resp.Header.Get("retry-after"))
```

Comment:
Discarding the error here is not so great as it leaves `retryAfter` set to `0`. Perhaps check for the error and set it to something like `60` by default to ensure that you don't end up in an eternal rate limiting loop?

Reply:
Ya, it's a good point. Arguably we should actually return an error in this case; if Slack responds with a 429 and doesn't set a retry-after error header, it's a malformed/unexpected response. That said, I think `10` is a reasonably sane failsafe, `60` is probably a bit too much. (Most of the rate limits are 1/second, but in this specific use case, it's 6/minute.)

Reply:
You are right of course, 60sec is a pretty insane `RetryAfter` value, but it's what Slack defaulted to last year. That said, I'm perfectly ok with a 10sec default too. :)